src/server.rs

```diff
@@ -51,6 +51,12 @@ pub async fn build(config: Config) -> anyhow::Result<SubwayServerHandle> {
 
     let rpc_method_weights = MethodWeights::from_config(&config.rpcs.methods);
 
+    // pre-check stage
+    if let Some(r) = &rate_limit_builder {
```

Comment:
if you want to check this then just make sure no weight is greater than burst (ip or connection)

Reply:
Yeah, The result is fixed if config is not changed, so we should do pre-check.